tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/openwire/SecurityOpenWireTest.java

```diff
@@ -16,70 +16,675 @@
+   }
+
+   @Test
+   public void testSendWithAuthToTopic() throws Exception {
+      doTestSendWithAuth(true);
+   }
+
+   private void doTestSendWithAuth(boolean topic) throws Exception {
+      try (Connection connection = factory.createConnection(ALLOWED_USER, PASS)) {
+         final Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+         final Destination destination;
+
+         if (topic) {
+            destination = session.createTopic(ADDRESS);
+         } else {
+            destination = session.createQueue(ADDRESS);
+         }
+
+         final MessageProducer producer = session.createProducer(destination);
+
+         producer.setDeliveryMode(DeliveryMode.PERSISTENT);
+
+         try {
+            producer.send(session.createMessage());
+            // Expected: can send to allowed destination
+         } catch (JMSSecurityException e) {
+            fail("unexpected error: " + e.getMessage());
+         }
+      }
+   }
+
+   @Test
+   public void testCreateSenderNoAuthToQueue() throws Exception {
+      doTestCreateSenderNoAuth(false);
+   }
+
+   @Test
+   public void testCreateSenderNoAuthToTopic() throws Exception {
+      doTestCreateSenderNoAuth(true);
+   }
+
+   private void doTestCreateSenderNoAuth(boolean topic) throws Exception {
+      try (Connection connection = factory.createConnection(DENIED_USER, PASS)) {
+         final Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+         final Destination destination;
+
+         if (topic) {
+            destination = session.createTopic(ADDRESS);
+         } else {
+            destination = session.createQueue(ADDRESS);
+         }
+
+         try {
+            session.createProducer(destination);
+            fail("Should not be able to create producer on restricted destination");
+         } catch (JMSSecurityException e) {
+            // expected to fail as not authorized
+         }
+      }
+   }
+
+   @Test
+   public void testCreateSenderWithAuthToFQQNAsQueue() throws Exception {
+      doTestCreateSenderWithAuthToFQQN(false);
+   }
+
+   @Test
+   public void testCreateSenderWithAuthToFQQNAsTopic() throws Exception {
+      doTestCreateSenderWithAuthToFQQN(true);
+   }
+
+   private void doTestCreateSenderWithAuthToFQQN(boolean topic) throws Exception {
+      try (Connection connection = factory.createConnection(ALLOWED_USER, PASS)) {
+         final Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+         final Destination destination;
+         final Destination otherDestination1;
+
+         if (topic) {
+            destination = session.createTopic(FQQN);
+            otherDestination1 = session.createTopic(FQQN_ALTERNATE); // Does not have authorization to this FQQN
+         } else {
+            destination = session.createQueue(FQQN);
+            otherDestination1 = session.createTopic(FQQN_ALTERNATE); // Does not have authorization to this FQQN
+         }
```

Comment:
otherDestination1 = session.createTopic(FQQN_ALTERNATE);

Is the same in both legs, intended? Can the 1 suffix be dropped?